lib/Models/BaseMaps/BaseMapsModel.ts

```diff
@@ -52,20 +51,26 @@ export class BaseMapsModel extends CreateModel(BaseMapsTraits) {
   get baseMapItems(): BaseMapItem[] {
     const enabledBaseMaps: BaseMapItem[] = [];
 
-    this.items.forEach((baseMapItem) => {
-      if (
-        baseMapItem.item &&
-        !ModelReference.isRemoved(baseMapItem.item) &&
-        (!this.enabledBaseMaps ||
-          this.enabledBaseMaps.includes(baseMapItem.item))
-      ) {
-        const itemModel = this.terria.getModelById(BaseModel, baseMapItem.item);
-        if (MappableMixin.isMixedInto(itemModel))
+    const baseMapItems =
+      this.enabledBaseMaps
+        ?.map((enabledBaseMapId) => {
+          return this.items.find((item) => item.item === enabledBaseMapId);
+        })
+        .filter(isDefined) ?? this.items;
+
+    baseMapItems.forEach((baseMapItem) => {
+      if (baseMapItem.item && !ModelReference.isRemoved(baseMapItem.item)) {
+        const itemModel = this.terria.getModelById(
+          BaseModel,
+          String(baseMapItem.item)
```

Comment:
Doubt: What's the reason we need to wrap with `String()` here?

Reply:
no reason, forgot to remove it